### PR TITLE
tests: adding unit tests

### PR DESCRIPTION
**Pull request by mgaseta**
# Description

As previously agreed, adding the unitary tests for the sample app :)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Adding the new tests, used `yarn test` to run them.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

Board link: https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-44

### CODE REVIEW

File: src/__test__/components/Card.test.tsx [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
-  expect(tree).toMatchSnapshot();
+describe('the component', () => {
+  it('should have the correct title and text', () => {
+    const { getByTestId } = render(<Card title="Test card" text="This is just a test text." />);
+    expect(getByTestId('card-title').textContent).toBe('Test card');
+    expect(getByTestId('card-text').textContent).toBe('This is just a test text.');
+  });
+});
+
+describe('the component renders correctly', () => {
```
Comment (dogivalfer) [inline src/__test__/components/Card.test.tsx:15]: I think this scenario should be merged with the previous one as every test case here is over the same component.

**Review by dogivalfer — CHANGES_REQUESTED** [review]
You can extend my suggestions for all the component tests that apply.

File: src/__test__/components/Card.test.tsx [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
 import React from 'react';
 import renderer from 'react-test-renderer';
 import Card from '../../components/Card';
 
-it('renders correctly', () => {
-  const tree = renderer
-    .create(<Card title="Test card" text="This is just a test text." />)
-    .toJSON();
-  expect(tree).toMatchSnapshot();
+describe('the component', () => {
```
[the remark this answers is not in the document]
Comment (mgaseta) [reply src/__test__/components/Card.test.tsx:7]: Added the component name to the describe :).

File: src/__test__/components/Card.test.tsx (hunk 1 above)
Comment (mgaseta) [reply src/__test__/components/Card.test.tsx:15]: Removed the second describe :)

File: src/__test__/components/TextInput.test.tsx [hunk 3]
```diff
@@ -1,24 +1,42 @@
 /* eslint-disable no-undef */
+import { render } from '@testing-library/react';
 import React from 'react';
 import renderer from 'react-test-renderer';
 import TextInput from '../../components/TextInput';
 
-describe('renders correctly', () => {
-  it('neutral input', () => {
+describe('the component', () => {
```
[the remark this answers is not in the document]
Comment (mgaseta) [reply src/__test__/components/TextInput.test.tsx:7]: Yep, added the changes :)

**Review by RMCampos — APPROVED** [review]
LGTM!

**Review by dogivalfer — APPROVED** [review]
LGTM!

**Pull request merged**